### ISSUE TITLE
Full Screen alignment/display option on several blocks

### ISSUE THREAD

**Issue by phpbits** [WordPress/gutenberg#16385]
**Is your feature request related to a problem? Please describe.**
Would be really nice to have ability to set several blocks as full screen. It's a very powerful option to have on Cover and Media Text Blocks.

Tagging @mtias :)

**Describe the solution you'd like**
I've added this feature on [EditorsKit](https://wordpress.org/plugins/block-options/) plugin and here's the preview : https://twitter.com/phpbits/status/1143122996552335362

**Comment by mtias** [comment]
I think this is pretty neat. It'd be interesting to see how this could work as an alignment option: "full screen". How do you handle that in the editor 100vh would be slightly more than what is visible (with the header toolbar, etc)?

**Comment by mtias** [comment]
@phpbits how do you feel about exploring how this might work within the alignments group?

**Comment by phpbits** [comment]
@mtias Sorry for late response. I'm actually planning to explore this are next week. I'm just finishing up the custom classes for editorsKit which is almost done. I'm really sure this will work on the alignment group though, I'm just not sure which filter to use right now. I'll let you know how it goes. Thanks!

**Comment by phpbits** [comment]
@mtias Just checked and it seems that there's now SlotFill available to register new alignment. It has to be manually added here : https://github.com/WordPress/gutenberg/blob/359858da0675943d8a759a0a7c03e7b3846536f5/packages/block-editor/src/components/block-alignment-toolbar/index.js . Thanks!

**Comment by mtias** [comment]
@phpbits yes, let's try to add it there directly as a new alignment and see how it feels.

**Comment by retrofox** [comment]
I'm going to work on this one. No guarantee, though.

**Comment by retrofox** [comment]
My first question is which icon would we use?
Are these ones all icons available?
https://developer.wordpress.org/resource/dashicons

**Comment by retrofox** [comment]
> let's try to add it there directly as a new alignment and see how it feels.

It seems to be possible to work combined with the alignment options. For instance, it's possible to align the image at the right and turn on the full-screen mode:

* align `right` - full screen `off`
![image](https://user-images.githubusercontent.com/77539/61658384-9fa7e680-ac9b-11e9-883f-b671b52cdf45.png)

* align `right` - full screen `off`
![image](https://user-images.githubusercontent.com/77539/61658248-58b9f100-ac9b-11e9-9199-bfe02befe5b1.png)

**Comment by phpbits** [comment]
@retrofox Is it possible to make the alignment extensible, similar to block styles? Thanks!

**Comment by retrofox** [comment]
hi @phpbits, 👋 
Could you help me to understand what you mean with extensible?

**Comment by retrofox** [comment]
> @retrofox Is it possible to make the alignment extensible, similar to block styles?

I think I got it. Do you mean to be able to add this option to other blocks, right? For instance, add it paragraphs, etc. Is it right?

**Comment by phpbits** [comment]
@retrofox I've actually had a second thought and removed my comment sorry. I just thought it'll be better if we can easily add alignment like `registerBlockStyle` but it should be different issue than this one.

**Comment by retrofox** [comment]
I've updated the PR this morning: https://github.com/WordPress/gutenberg/pull/16738 👋

**Comment by kjellr** [comment]
Just noting that I'd love to see this resurrected. A fullscreen alignment would be excellent for use in certain block patterns.

**Comment by mtias** [comment]
Now that we have vh unit support at the component level it seems we could make the alignment be a shortcut to it. cc @ItsJonQ

**Comment by retrofox** [comment]
> Now that we have vh unit support at the component level it seems we could make the alignment be a shortcut to it. cc @ItsJonQ

tackled here https://github.com/WordPress/gutenberg/pull/26615

**Comment by retrofox** [comment]
re-open...

**Comment by melchoyce** [comment]
Can we add `vh` to the Media & Text block as well?

**Comment by retrofox** [comment]
> Can we add `vh` to the Media & Text block as well?

something talked with @jasmussen  too if I'm not wrong. Thoughts?

**Comment by jasmussen** [comment]
> Can we add vh to the Media & Text block as well?

It looks like the first step would be to add a general height control to the Media & Text block, that's not present yet.
